projects/js-packages/ai-client/src/chrome-ai/get-availability.ts

```diff
@@ -31,6 +32,45 @@ function getAiAssistantFeature() {
 	return getFeature();
 }
 
+/**
+ * Fetch an experiment assignment.
+ *
+ * @param {boolean} asConnectedUser - Whether the user is connected.
+ * @return {Function} A function that fetches an experiment assignment.
+ */
+const fetchExperimentAssignment =
+	( asConnectedUser = false ) =>
+	async ( {
+		experimentName,
+		anonId,
+	}: {
+		experimentName: string;
+		anonId: string | null;
+	} ): Promise< unknown > => {
+		if ( ! anonId ) {
+			debug( 'anonId is null' );
+			throw new Error( `Tracking is disabled, can't fetch experimentAssignment` );
+		}
+
+		const params = {
+			experiment_name: experimentName,
+			anon_id: anonId ?? undefined,
```

Comment:
For reference, the `/assignments` endpoint requires one more IDs:
- If `anon_id` only or `userid` only is provided, then a variation assignment can be successfully returned.
- If both `anon_id` and `userid` is provided, then the assignment endpoint prioritizes `anon_id` over `userid`.

For this experiment, since the logged-in userid should be available, we should **omit `anon_id`** when making the `/assignments` request, ensuring that assignment is based solely on `userid`.